### PR TITLE
Enable BPF on ARM64

### LINKED ISSUE #14206: BPF doesn't compile on arm64
Expected behavior:
BPF code should compile and run on ARM64 platforms. 

Current behavior:

BPF code failed to compile on ARM64.

Bug details:
- Teleport version v10 but probably some earlier releases too.
- Recreation steps
- Debug logs

### PR DESCRIPTION
**Pull request by jakule**
Enable BPF build on ARM64

Closes https://github.com/gravitational/teleport/issues/14206

TODO:
- [x] Merge https://github.com/gravitational/teleport.e/pull/890

### CODE REVIEW

File: build.assets/Dockerfile-centos7 [hunk 1]
```diff
@@ -1,17 +1,49 @@
-ARG RUST_VERSION
+## GIT2 ###################################################################
```
Comment (reedloden) [inline build.assets/Dockerfile-centos7:1]: I think `Dockerfile-centos7-fips` needs these changes as well

File: build.assets/Dockerfile-centos7 [hunk 2]
```diff
@@ -247,7 +285,9 @@ COPY --from=libpcsclite \
     /usr/local/lib/
 
 # Download pre-built CentOS 7 assets with clang needed to build BPF tools.
-RUN cd / && curl -L https://s3.amazonaws.com/clientbuilds.gravitational.io/go/centos7-assets.tar.gz | tar -xz
+#RUN cd / && curl -L https://s3.amazonaws.com/clientbuilds.gravitational.io/go/centos7-assets.tar.gz | tar -xz
```
Comment (reedloden) [inline build.assets/Dockerfile-centos7:288]: Yesssss. Get rid of this with all haste, please. We pay a lot of $$$ every month for this thing getting downloaded constantly.

Comment (jakule) [reply build.assets/Dockerfile-centos7:288]: @reedloden I think we can keep the "centos assets" as a container in GitHub. I just need to coordinate this with someone.

File: build.assets/Dockerfile-centos7 (hunk 1 above)
Comment (jakule) [reply build.assets/Dockerfile-centos7:1]: Thanks. I forgot about it. I feel like our Dockerfiles multiplies every few months. I'll try to clean them up later, as I think we don't use most of them.

**Review by zmb3 — APPROVED** [review]

**Comment by jakule** [comment]
@tcsc @r0mant @smallinsky @greedy52 Friendly ping

**Review by greedy52 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**